programmers/제일-작은-수-제거하기.js

```diff
@@ -0,0 +1,9 @@
+function solution(arr) {
+    const arrMin = Math.min(...arr);
+    const i = arr.findIndex(num => num === arrMin);
+    arr.splice(i, 1);
+    if (arr.length <= 1) {
+        return [-1];
+    }
+    return arr;
+}
```

Comment:
1. findIndex 메서드 자체가 파라미터로 `함수(콜백 함수)`를 받음, 그리고 그 `콜백 함수`의 조건에 맞는 Index를 반환해줌.
예로들면, arr이 [4,. 3, 2, 1]인 경우 num은  4 -> 3 -> 2 -> 1 이렇게 순환하는데 그중에 arrMin과 동일한 값. 즉, 1의 index(3)를 return 함

+a) 함수에서 `{ }` 를 제거하고 한줄로 작성할 경우 return 키워드를 사용하지 않아도 바로 return 됨. 반대로 `{ }` 를 넣으면 return이 필요한 경우 꼭 return을 넣어줘야 함
```js
    const minIdx = arr.findIndex(num => {
        console.log(num); //4->3->2->1
        return num === MIN_NUM
    });
```

2. arrMin은 MIN_NUM 변수명이 나아보임

3. i보단 좀 더 명확한 변수명이 나아보임

```suggestion
function solution(arr) {
    const MIN_NUM= Math.min(...arr);
    const minIdx = arr.findIndex(num => num === MIN_NUM);
    
    arr.splice(minIdx , 1);
    
    if (arr.length <= 1) return [-1];
    return arr;
}
```

Reply:
findIndex 메서드 대신 indexOf 메서드를 사용해도 됨
```js
function solution(arr) {
    const MIN_NUM = Math.min(...arr);
    const minIdx = arr.indexOf(MIN_NUM);
    
    arr.splice(minIdx, 1);
    
    if (arr.length <= 1) return [-1];
    return arr;
}
```